crates/algo_models/src/lib.rs

```diff
@@ -100,6 +98,27 @@ pub trait AlgorandMsgpack: Serialize + for<'de> Deserialize<'de> {
     }
 }
 
+pub trait TransactionId: AlgorandMsgpack {
+    fn raw_id(&self) -> Result<[u8; HASH_BYTES_LENGTH], AlgoModelsError> {
+        let mut hasher = Sha512_256::new();
+        hasher.update(self.encode()?);
+        hasher.finalize().try_into().map_err(|_| {
+            AlgoModelsError::InputError(
```

Comment:
Was tossing up if I should create a new error type. It should never be hit.

Reply:
Since hash length is an invariant I would say this would be an acceptable use of `except` instead of `map_err`.